python-checks/src/main/java/org/sonar/python/checks/cdk/AbstractCdkResourceCheck.java

```diff
@@ -171,4 +176,42 @@ protected static Predicate<Expression> isFqn(String fqnValue) {
       .isPresent();
   }
 
+  protected static Predicate<Expression> isString(String str) {
+    return expression ->  Optional.of(expression)
+      .filter(expr -> expr.is(Tree.Kind.STRING_LITERAL)).map(StringLiteral.class::cast)
+      .filter(stringLiteral -> stringLiteral.trimmedQuotesValue().equals(str))
+      .isPresent();
+  }
+
+  protected static Predicate<Expression> isDictionaryWithKeyValue(SubscriptionContext ctx, String key, String val) {
+    return expression -> expression.is(Tree.Kind.DICTIONARY_LITERAL)
+      && asDictionaryKeyValuePairs(expression)
+      .filter(isKey(key))
+      .allMatch(isKeyValueString(ctx, val));
+  }
+
+  private static Stream<KeyValuePair> asDictionaryKeyValuePairs(Expression expression) {
+    return asDictionaryElements(expression)
+      .filter(element -> element.is(Tree.Kind.KEY_VALUE_PAIR)).map(KeyValuePair.class::cast);
+  }
+
+  private static Stream<DictionaryLiteralElement> asDictionaryElements(Expression expression) {
+    return Optional.of(expression)
+      .filter(expr -> expr.is(Tree.Kind.DICTIONARY_LITERAL)).map(DictionaryLiteral.class::cast)
+      .map(DictionaryLiteral::elements)
+      .orElse(Collections.emptyList()).stream();
+  }
+
+  private static Predicate<KeyValuePair> isKey(String keyName) {
+    return keyValuePair -> Optional.of(keyValuePair)
+      .map(KeyValuePair::key)
+      .filter(key -> key.is(Tree.Kind.STRING_LITERAL)).map(StringLiteral.class::cast)
+      .filter(string -> string.trimmedQuotesValue().equals(keyName))
+      .isPresent();
+  }
+
+  private static Predicate<KeyValuePair> isKeyValueString(SubscriptionContext ctx, String str) {
+    return keyValuePair -> ArgumentTrace.build(ctx, keyValuePair.value()).hasExpression(isString(str));
+  }
```

Comment:
You should have a look at `org.sonar.python.checks.cdk.ClearTextProtocolsCheckPart.ResolvedKeyValuePair` and its use. Maybe you can make use of some logic to reduce duplications.